src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,400 @@
+    typealias HeatmapInterpolationInfluence = Double
+
+    /// Indicates the number of times k-means clustering should execute; will be set in the constructor to 25 by default
+    private var clusterIterations: Int!
+
+    /// Normalizing factors to convert from 2D to longitude and latitude; these values were needed since each GMUWeightedLatLng
+    /// point contains a .point() field, which is a GQTPoint. The point value for the GQTPoint should be equivalent to a GMSMapPoint,
+    /// with range [-1.0, 1.0]. I found that multiplying the latitude and longitude by the following numbers correctly converts the
+    /// GQTPoint value (which was all from -1.0 to 1.0) to the latitude and longitude input (as CLLocationCoordinate2D).
+    private let normalLat = 175.9783070993
+    private let normalLong = 180.0
+
+    /// Firm bounds on all search queries, as latitude ranges from -90 to 90 and longitude ranges from -180 to 180
+    private let minLat = -90
+    private let maxLat = 90
+    private let minLong = -180
+    private let maxLong = 180
+
+    /// The constructor to this class
+    ///
+    /// - Parameter givenClusterIterations: The number of iterations k-means clustering should go to.
+    init(givenClusterIterations: Int = 25) {
+        clusterIterations = givenClusterIterations
+    }
+
+    // MARK: Functions that parse given data needed to build an interpolated heat map from
+
+    /// Adds a list of GMUWeightedLatLng objects to the input data set
+    ///
+    /// - Parameter latlngs: The list of GMUWeightedLatLng objects to add.
+    public func addWeightedLatLngs(latlngs: [GMUWeightedLatLng]) {
+        data.append(contentsOf: latlngs)
+    }
+
+    /// Adds a single GMUWeightedLatLng object to the input data set
+    ///
+    /// - Parameter latlngs: The list of GMUWeightedLatLng objects to add.
+    public func addWeightedLatLng(latlng: GMUWeightedLatLng) {
+        data.append(latlng)
+    }
+
+    /// Removes all previously supplied GMUWeightedLatLng objects
+    public func removeAllData() {
+        data.removeAll()
+    }
+
+    /// Throw this error when the given influence value is out of range (i.e. not between 2 and 2.5)
+    enum IncorrectInfluence: Error {
+        case outOfRange(String)
+    }
+
+    // MARK: Functions that directly contribute to the creation of interpolated points
+
+    /// A helper function that calculates the straight-line distance between two coordinates
+    ///
+    /// - Parameters:
+    ///   - lat1: The latitude value of the first point.
+    ///   - long1: The longitude value of the second point.
+    ///   - lat2: The latitude value of the second point.
+    ///   - long2: The longitude value of the second point.
+    /// - Returns: A double value representing the distance between the given points.
+    private func distance(lat1: Double, long1: Double, lat2: Double, long2: Double) -> Double {
```

Comment:
Instead of using `Double`, use `CLLocationDegrees` instead so that the type is clearer to the reader.